tools/src/test/java/com/example/core/broker/testkit/services/DeltaDownstreamServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.example.core.broker.testkit.services;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.stream.javadsl.Flow;
+import com.lightbend.lagom.javadsl.api.ServiceCall;
+import org.pcollections.PSequence;
+import org.pcollections.TreePVector;
+
+import javax.inject.Inject;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+public class DeltaDownstreamServiceImpl implements DeltaDownstreamService {
+
+    private final List<ReceivedMessage> receivedMessages = new ArrayList<>();
```

Comment:
I think there could be a thread safety issue using an unsynchronized `ArrayList` here. May be better to use a volatile, mutable reference to a `TreePVector` the whole way through, or alternatively an immutable reference to a mutable `ConcurrentLinkedQueue`.